torch/utils/cpp_extension.py

```diff
@@ -2422,3 +2606,524 @@ def _is_cuda_file(path: str) -> bool:
     if IS_HIP_EXTENSION:
         valid_ext.append('.hip')
     return os.path.splitext(path)[1] in valid_ext
+
+def get_dpcpp_complier():
```

Comment:
the dpcpp name will confuse latecomers, as we do not use it in all xpu upstreamed code now, even the name of the compiler itself.

@guangyey  @EikanWang  Any suggestion here?

Reply:
That's a good point.

dpcpp is actually one of few sycl compilers available even right now. Who knows, maybe in the future, it might be potentially possible to use other compilers as well? And dpcpp has this mark that "we try to upstream as much DPC++ support to LLVM main branch as possible" (https://github.com/intel/llvm) implying that clang might get required capabilities at some point later.

Maybe we should consider to rename stuff to be "sycl" compiler rather than go with dpcpp anywhere?

Reply:
We should not mention `dpcpp` in our code anymore.